nltk/probability.py

```diff
@@ -1954,8 +1956,9 @@ def tabulate(self, *args, **kwargs):
         cumulative = _get_kwarg(kwargs, 'cumulative', False)
         conditions = _get_kwarg(kwargs, 'conditions', sorted(self.conditions()))
         samples = _get_kwarg(
-            kwargs, 'samples', sorted(set(v for c in conditions for v in self[c]))
-        )  # this computation could be wasted
+            kwargs, 'samples', sorted(set(v for c in conditions
+                                          if c in self
+                                          for v in self[c])))  # this computation could be wasted
```

Comment:
Looks right but we've to test cases where this will happen.

Reply:
Oops! No idea how that typo got in there. I will fix that and write a short test in the next day or two :)